michbl/assignment2/BinaryTree.hpp

```diff
@@ -0,0 +1,194 @@
+#ifndef BINARYTREE_HPP
+#define BINARYTREE_HPP
+
+#include <iostream>
+using namespace std;
+
+// Assuming T is comparable and printable

```

Comment:
It is good that you documented what are the preconditions of T.
This class also assumes that T is copyable :)